src/node_kernels/MomentumMassBDFNodeKernel.C

```diff
@@ -0,0 +1,93 @@
+/*------------------------------------------------------------------------*/
+/*  Copyright 2014 Sandia Corporation.                                    */
+/*  This software is released under the license detailed                  */
+/*  in the file, LICENSE, which is located in the top-level Nalu          */
+/*  directory structure                                                   */
+/*------------------------------------------------------------------------*/
+
+
+#include "node_kernels/MomentumMassBDFNodeKernel.h"
+#include "Realm.h"
+
+#include "stk_mesh/base/MetaData.hpp"
+#include "utils/StkHelpers.h"
+
+namespace sierra{
+namespace nalu{
+
+MomentumMassBDFNodeKernel::MomentumMassBDFNodeKernel(
+  const stk::mesh::BulkData& bulk,
+  ScalarFieldType *scalarQ
+) : NGPNodeKernel<MomentumMassBDFNodeKernel>(),
+    nDim_(bulk.mesh_meta_data().spatial_dimension())
+{
+  const auto& meta = bulk.mesh_meta_data();
+
+  velocityNID_ = scalarQ->field_of_state(stk::mesh::StateN).mesh_meta_data_ordinal();
```

Comment:
This is incorrect, `velocity` is a `VectorFieldType` and this line should be 

```
velocityNID_ = get_field_ordinal(meta, "velocity", stk::mesh::StateN);
```

Similarly for the remaining states.